packages/unified-server/src/shell/opal-shell-host.ts

```diff
@@ -20,21 +20,30 @@ class OpalShellProtocolImpl implements OpalShellProtocol {
   }
 }
 
-function expose() {
+const guestOrigin = CLIENT_DEPLOYMENT_CONFIG.SHELL_GUEST_ORIGIN;
+if (guestOrigin && guestOrigin !== "*") {
   const iframe = document.querySelector("iframe#opal-app" as "iframe");
-  if (!iframe?.contentWindow) {
+  if (iframe?.contentWindow) {
+    comlink.expose(
+      new OpalShellProtocolImpl(),
+      comlink.windowEndpoint(
+        // Where this host sends messages.
+        iframe.contentWindow,
+        // Where this host receives messages from.
+        window,
+        // Constrain origins this host can communicate with, at the postMessage
+        // layer. It would otherwise default to all origins.
+        // https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#targetorigin
+        guestOrigin
+      ),
+      // Constrain origins this host can communicate with, at the comlink layer.
+      // It would otherwise default to all origins. Note this is technically
```

Comment:
Nit: It's not redundant right? The postMessage restriction constrains which domains the guest will expose data to. The onMessage restriction constrains which domains the host will trust messages from.